lib/luaT/luaT.c

```diff
@@ -497,27 +496,65 @@ int luaT_classmodulename(const char *tname, char *module_name)
   return tname[n] == '.';
 }
 
-/* Lua only functions */
+void traverse_tables(lua_State *L, const char *tname, int *L_idx)
```

Comment:
I would imagine that passing in the index is more efficient; as it avoid either an extra call to lua_gettop() or use of relative indices (e.g. -1). That said, I can certainly update it if preferred.